inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp

```diff
@@ -1205,52 +1210,64 @@ void MKLDNNEltwiseNode::initSupportedPrimitiveDescriptors() {
 
     bool isBlockedApplicable = one_of(getOutputShapeAtPort(0).getRank(), 1, 4, 5);
     for (size_t i = 0; i < getParentEdges().size(); i++) {
-        isBlockedApplicable = isBlockedApplicable && one_of(getInputShapeAtPort(i).getRank(), 1, 4, 5);
-        isBlockedApplicable = isBlockedApplicable && implication(getInputShapeAtPort(i).getRank() != 1,
+        const auto &inShape = getInputShapeAtPort(i);
+        isBlockedApplicable = isBlockedApplicable && one_of(inShape.getRank(), 1, 4, 5);
+        isBlockedApplicable = isBlockedApplicable && implication(inShape.getRank() != 1,
                                                                  getOutputShapeAtPort(0).getRank() ==
-                                                                 getInputShapeAtPort(i).getRank());
+                                                                 inShape.getRank());
+        if (isDynamicNode() && inShape.getRank() != 1)
+            isBlockedApplicable = isBlockedApplicable && inShape.getMinDims()[1] != Shape::UNDEFINED_DIM && inShape.getMinDims()[1] > 1;
     }
 
     if (isChannelsFirstApplicable)
         supportedPrimitiveDescriptors.emplace_back(initDesc(ChannelsFirst));
     if (isBlockedApplicable)
         supportedPrimitiveDescriptors.emplace_back(initDesc(Blocked));
     supportedPrimitiveDescriptors.emplace_back(initDesc(Planar));
-}
 
-void MKLDNNEltwiseNode::createPrimitive() {
-    auto config = getSelectedPrimitiveDescriptor()->getConfig();
     inputNum = getParentEdges().size();
+    currentInBlkDims.resize(inputNum);
+    currentInDims.resize(inputNum);
```

Comment:
Looks like an error prone solution. It is really easy to forget resize this vector. We have to provide a better solution.

Reply:
let's discuss it offline

my current solution to add additional pass into `InitGraph` only for dynamic cases. It's necessary because we add additional input  edges in some plug-in transformation

Reply:
extract in separate method and pass in `InitGraph`